diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt

```diff
@@ -135,11 +122,79 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
             BLOCK -> handleLambdaBody(node)
             RETURN -> handleReturnStatement(node)
             SUPER_TYPE_LIST, VALUE_PARAMETER_LIST, VALUE_ARGUMENT_LIST -> handleList(node)
+            DOT_QUALIFIED_EXPRESSION, SAFE_ACCESS_EXPRESSION, POSTFIX_EXPRESSION -> handDotQualifiedAndSafeAccessExpression(node)
             else -> {
             }
         }
     }
 
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION", "MagicNumber")
+    private fun handDotQualifiedAndSafeAccessExpression(node: ASTNode) {
+        val listParentTypesNoFix = listOf<IElementType>(PACKAGE_DIRECTIVE, IMPORT_DIRECTIVE, VALUE_PARAMETER_LIST,
+            VALUE_ARGUMENT_LIST, DOT_QUALIFIED_EXPRESSION, SAFE_ACCESS_EXPRESSION, POSTFIX_EXPRESSION)
+        if (isNotFindParentNodeWithSpecificManyType(node, listParentTypesNoFix)) {
+            val listDot: MutableList<ASTNode> = mutableListOf()
+            searchDot(node, listDot)
+            if (listDot.size > 3) {
+                val (withRightWhiteSpace, without) = listDot.partition {
```

Comment:
`withRightWhiteSpace` is not used, `partition` can be replaced by `filterNot`

Reply:
Done [05c5a44](https://github.com/saveourtool/diktat/pull/1337/commits/05c5a44142c2b44fb2db5e8663c3d1bb8c8a4247)